tools/test.py

```diff
@@ -172,7 +172,10 @@ def Run(self, tasks):
       # ...and then reraise the exception to bail out
       raise
     self.Done()
-    return not self.failed
+    return {
+      'allPassed': not self.failed,
+      'failed': self.failed,
+    }
```

Comment:
Let's make sure that method signature specifies its return type.

Currently, it would be `def Run(self, tasks) -> bool:` and with the proposed change, it would become `def Run(self, tasks) -> dict:`.

Perhaps, an even better approach would be to just return `self.failed` because the caller can trivially calculate `all_passed`.  The signature would be `def Run(self, tasks) -> list[TestResult]:`

Reply:
> Perhaps, an even better approach would be to just return self.failed because the caller can trivially calculate all_passed. The signature would then be def Run(self, tasks) -> list[TestResult]:

I considered this, but I think it would be safer to be more explicit. The old caller assumes that true means everything passed; if I hadn't updated the caller, the calling code would continue to work but behave the opposite as expected (where now "evaluates as false" means that they all passed). I don't know if there are other callers to this function such as potentially in user land, like other node devs writing their custom extensions of this script; with the dict return, such code would definitively error with a message that would point them to this change.

Reply:
Type hints are explicit.  If you are worried about other callers then do not change the function.

Reply:
I’ll add the type hint. I still think the dict return is both safer and more future-proof, for example if we want to add a list of passing tests or any other output for an extended summary.